Source/WebCore/Modules/webdatabase/SQLTransactionStateMachine.h

```diff
@@ -98,8 +98,9 @@ void SQLTransactionStateMachine<T>::runStateMachine()
     ASSERT(stateFunction);
 
 #ifndef NDEBUG
-    m_stateAuditTrail[m_nextStateAuditEntry] = m_nextState;
-    m_nextStateAuditEntry = (m_nextStateAuditEntry + 1) % s_sizeOfStateAuditTrail;
+    m_stateAuditTrail[m_nextStateAuditEntry++] = m_nextState;
+    if (m_nextStateAuditEntry == s_sizeOfStateAuditTrail)
+        m_nextStateAuditEntry = 0;
```

Comment:
I see this change is equivalent, but it's also harder to read than the original code. I highly doubt this is going to perform any better, and it's only built in debug builds anyway. I wouldn't change this.